code/obj/submachine/cooking.dm

```diff
@@ -1573,53 +1576,63 @@ TYPEINFO(/obj/submachine/mixer)
 		user.u_equip(W)
 		W.set_loc(src)
 		W.dropped(user)
+		tgui_process.update_uis(src)
+
+	ui_interact(mob/user, datum/tgui/ui)
+		ui = tgui_process.try_update_ui(user, src, ui)
+		if(!ui)
+			ui = new(user, src, "MixerMachine")
+			ui.open()
+	ui_static_data(mob/user)
+		. = list("maxItems" = MIXER_MAX_CONTENTS)
+
+	ui_data(mob/user)
+		var/mixerContents = list()
+		var/index = 1
+		for(var/obj/item/_Item in src.contents)
+			var/itemData = list()
+			itemData["name"] = _Item.name
+			itemData["index"] = index
+			mixerContents += list(itemData)
+			index += 1
+		. = list("working" = src.working, "mixerContents" = mixerContents)
+
+	proc/ejectItemFromMixer(obj/item/target)
+		if (target)
+			if (BOUNDS_DIST(usr, src) == 0)
+				usr.put_in_hand_or_drop(target)
+			else
+				target.set_loc(src.loc)
 
-	attack_hand(var/mob/user)
-		if (!src.working)
-			src.add_dialog(user)
-			var/dat = {"<B>KitchenHelper Mixer</B><BR>
-			<HR>
-			<B>Contents:</B><BR>"}
-			for (var/obj/item/I in src.contents)
-				dat += "[I]<BR>"
-			dat += {"<HR>
-			<A href='?src=\ref[src];mix=1'>Mix!</A><BR>
-			<A href='?src=\ref[src];eject=1'>Eject Contents</A>"}
-			user.Browse(dat, "window=mixer;size=400x500")
-			onclose(user, "mixer")
-		else
-			src.add_dialog(user)
-			var/dat = {"<B>KitchenHelper Mixer</B><BR>
-			<HR><BR>
-			Mixing! Please wait!"}
-			user.Browse(dat, "window=mixer;size=400x500")
-			onclose(user, "mixer")
+	ui_act(action, params)
+		. = ..()
+		if (.)
+			return
 
-	attack_ai(var/mob/user as mob)
-		return attack_hand(user)
+		switch (action)
+			if ("eject")
+				var/index = params["index"]
+				var/obj/item/target = src.contents[index]
+				src.ejectItemFromMixer(target)
+
+				usr.show_text(SPAN_NOTICE("You ejected the [target.name] from the [src]."))
```

Comment:
These messages usually use present-tense, ie "You eject" rather than "You ejected" but also they're proobably not needed here since you know what you just clicked on.

Reply:
Fixed